File: run_mop.R
source("process_data.R")
source("methods.R")

args <- commandArgs(trailingOnly = TRUE)

sample_id <- args[[1]] # e.g., "mouse1_sample1"
method <- args[[1]] # e.g., DOT, RCTD, SPOTlight, CARD, ...

experiment_data <- produce_mop_multicell(sample_id)
annotations <- experiment_data$ref$meta[, "cluster_label"]
locations <- experiment_data$srt$meta[, c('x', 'y')]

if(method == "DOT"){
  weights <- run_DOT(experiment_data$ref$counts, annotations,
                     experiment_data$srt$counts, locations)
}else if(method == "RCTD"){
  weights <- run_RCTD(experiment_data$ref$counts, annotations,
                      experiment_data$srt$counts, locations)
}else if(method == "SPOTlight"){
  weights <- run_SPOTlight(experiment_data$ref$counts, annotations,
                      experiment_data$srt$counts, n_cells = 100)
}else if(method == "CARD"){
  weights <- run_CARD(experiment_data$ref$counts, annotations,
                     experiment_data$srt$counts, locations)
}else if(method == "SingleR"){
  weights <- run_SingleR(experiment_data$ref$counts, annotations,
                        experiment_data$srt$counts)
}else if(method == "Seurat"){
  weights <- run_Seurat(experiment_data$ref$counts, annotations,
                         experiment_data$srt$counts)
}else if(method == "Harmony"){
  weights <- run_Harmony(experiment_data$ref$counts, annotations,
                         experiment_data$srt$counts)
}else if(method == "RF"){
  weights <- run_RF(experiment_data$ref$counts, annotations,
                         experiment_data$srt$counts)
}

File: run_mop.py
import sys
from process_data import produce_mop_multicell

args = sys.argv
sample_id = args[1] # e.g., "mouse1_sample1"
method = args[2] # e.g., TACCO, cell2location, ...

ref_adata, srt_adata = produce_mop_multicell(sample_id)
ref_annotation = "cluster_label"

if method.lower() == "tacco":
    from methods import run_tacco
    weights = run_tacco(ref_adata, srt_adata, ref_annotation)
elif method.lower() in ["cell2location", "c2l"]:
    from methods import run_cell2location
    weights = run_cell2location(ref_adata, srt_adata, ref_annotation, epochs = 20000)
elif method.lower() == "tangram":
    from methods import run_tangram
    weights = run_tangram(ref_adata, srt_adata, ref_annotation)
elif method.lower() == "novosparc":
    from methods import run_novosparc
    coords = srt_adata.obs[['x', 'y']]
    weights = run_novosparc(ref_adata, srt_adata , ref_annotation, coords, alpha = 0.5, ncells = 500)